drivers/serial/uart_nrfx_uart.c

```diff
@@ -335,17 +335,37 @@ static int uart_nrfx_init(struct device *dev)
 		 "UART init failed. Cannot find %s",
 		 CONFIG_GPIO_NRF5_P0_DEV_NAME);
 
-	(void) gpio_pin_configure(gpio_dev,
-				  CONFIG_UART_0_NRF_TX_PIN,
-				  (GPIO_DIR_OUT | GPIO_PUD_PULL_UP));
-	(void) gpio_pin_configure(gpio_dev,
-				  CONFIG_UART_0_NRF_RX_PIN,
-				  (GPIO_DIR_IN));
+	/* Setting default height state of the TX PIN to avoid glitches
+	 * on the line during peripheral activation/deactivation.
+	 */
+	gpio_pin_write(gpio_dev, CONFIG_UART_0_NRF_TX_PIN, 1);
```

Comment:
this should be after the pin is configured as output? or is it ok before, so that when configured as output then there is no glitch?

Reply:
Yes, this is the intention to avoid glitch. 

In fact this is how it was working so far.
When gpio_nrf5 driver detected "pull up" flag it was setting high state on the PIN and next it was configuring PIN as output.